### ISSUE TITLE
Compliance with the JDK's Standard Mechanism for Configuring Service Providers

### ISSUE THREAD

**Issue by lingocoder** [gradle/gradle#11027]
### Expected Behavior

The Gradle build tool is expected to adhere to the standard mechanisms and correct usage of the services that are specified by the JDK. Specifically, it's expected that Gradle complies with and correctly configures the JDK-specified service provider locator facilities it uses.

### Current Behavior

In Gradle version 5.6.2, the *`«$GRADLE_USER_HOME»/caches/5.6.2/generated-gradle-jars/gradle-api-5.6.2.jar`* file contains a service configuration file that — *according to current JDK documentation* — does not comply with the standard mechanisms of usage specified by the JDK documentation. The same non-compliant configuration can currently be found in Gradle's version 6+ early access releases too.

### Context

As a developer of a library that will be distributed as a JPMS module, I am trying to accomplish unit testing of the classes of my JPMS-modularized library.

During the development and unit testing of my library, I was affected by the JPMS development hurdle reported in Gradle issue #10408. 

The specific JDK-specified standard mechanism that Gradle is not in compliance with, is prescribed in [*the JDK's ServiceLoader API documentation*](http://bit.ly/J12SvcLoadr)...

> ...
>
>  ***Deploying service providers on the class path*** 
> ...
> *The service provider must be visible from the class loader that is initially queried to locate the provider-configuration file...*
> ...
>

### The Problem

***TL;DR*** — The gradle-api-{version}.jar file lists a non-existent service provider in its *`META-INF/services/java.security.Provider`* provider-configuration file. A non existent service provider cannot „*be visible from the class loader*“ — which the JDK documentation prescribes as a „*must*“.

This jigsaw-dev mailing list message from @AlanBateman — *a co-author of and senior contributor to the Jigsaw Project* — offers some insight into why ignoring the problem is risky:

> ...
> „*the analysis in [issue #10408] seems to be correct. Gradle*
> *seems to have re-packaged the BC provider into the* 
> *org.gradle.internal.impldep.org.bouncycastle name space but missed the* 
> *services configuration file. There could [be] issues here on multiple levels* 
> *that are nothing to do with modules....*“
> 
> -Alan
> ...

A more detailed discussion of the problem can be found [*here*](http://bit.ly/jigsawdev014289).

### Steps to Reproduce 

1. [*Download and unpack this zip-archived project*](http://bit.ly/antiUpLoad)
2. In a terminal opened in the root folder of the downloaded project, run the *`:test`* task:

```
./gradlew :test
```
3. Observe the error printed to stdout:

```
...
...[QUIET] [system.out] Error occurred during initialization of boot layer
...[QUIET] [system.out] java.lang.module.FindException: Unable to derive module descriptor for «$GRADLE_USER_HOME»/caches/5.6.2/generated-gradle-jars/gradle-api-5.6.2.jar
...[QUIET] [system.out] Caused by: java.lang.module.InvalidModuleDescriptorException: Provider class org.bouncycastle.pqc.jcajce.provider.BouncyCastlePQCProvider not in module
...
```

### Your Environment

```
------------------------------------------------------------
Gradle 5.6.2
------------------------------------------------------------

Build time:   2019-09-05 16:13:54 UTC
Revision:     55a5e53d855db8fc7b0e494412fc624051a8e781

Kotlin:       1.3.41
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          12 (Oracle Corporation 12+33)
OS:           Windows amd64
```

**Comment by lingocoder** [comment]
>> *...When you have five minutes to spare, please, can you point me in the direction of where to find the file (and/or related code) that needs to be changed for this one?...*

[*Found it*](https://github.com/gradle/gradle/pull/11093), @melix.

**Comment by rgoldberg** [comment]
Is this issue the same as #10408?

If so, that has been fixed, and this should also be able to be closed.

If it is different than #10408, can someone please explain the difference?
